pkg/trustedresources/verify.go

```diff
@@ -41,6 +41,28 @@ const (
 	SignatureAnnotation = "tekton.dev/signature"
 )
 
+const (
+	VerificationSkip = iota
+	VerificationPass
+	VerificationWarn
+	VerificationError
+)
+
+// VerificationResultType indicates different cases of a verification result
+type VerificationResultType int
+
+// VerificationResult contains the type and message about the result of verification
+type VerificationResult struct {
+	// VerificationResultType has 4 types which is corresponding to 4 cases:
+	// 1) VerificationSkip: The verification was skipped. Err is nil in this case.
+	// 2) VerificationPass: The verification passed. Err is nil in this case.
+	// 3) VerificationWarn: A warning is logged. It could be no matching policies and feature flag "no-match-policy" is "warn", or only Warn mode verification policies fail.
+	// 4) VerificationError: The verification failed, it could be the signature doesn't match the public key, no matching policies and "no-match-policy" is set to "fail" or there are errors during verification.
```

Comment:
Just minor comment: can we use the corresponding int for each mode?

```suggestion
	// 0 (VerificationSkip): The verification was skipped. Err is nil in this case.
	// 1 (VerificationPass): The verification passed. Err is nil in this case.
	// 2 (VerificationWarn): A warning is logged. It could be no matching policies and feature flag "no-match-policy" is "warn", or only Warn mode verification policies fail.
	// 3 (VerificationError): The verification failed, it could be the signature doesn't match the public key, no matching policies and "no-match-policy" is set to "fail" or there are errors during verification.
```

Reply:
Thanks! Good suggestion!